apps/labs/posts/numba-polynomial-support.md

```diff
@@ -0,0 +1,64 @@
+
+## Adding support for Power Series in Numba
+Similarly to functions, NumPy classes have to be reimplemented in Numba using its low-level API. A good example of this can be found in the [Numba documentation](https://numba.readthedocs.io/en/stable/extending/interval-example.html#example-an-interval-type), which I followed very closely. The first big task when working on the `Polynomial` class was defining a new Numba type to represent the class. After creating a type class called `PolynomialType`, I needed to "show" Numba that each instance of the `Polynomial` class should be treated by Numba as if its type is `PolynomialType`, using the `@typeof_impl.register(np.polynomial.polynomial.Polynomial)` decorator. Another important part is defining the constructor for the Polynomial class. Me and my mentor decided to add support for `Polynomial(coef)` and `Polynomial(coef, domain, window)` at this point. As a general rule, nopython mode does not work on Python objects directly. Each type supported in nopython mode therefore has to define a tailored native representation, also called a data model.
+
+The final task when adding support for an external class in Numba is defining special methods for converting between the NumPy object and the Numba data model. Those are called boxing and unboxing, the idea being that a Python object can be seen as a sophisticated box containing a simple native value which is what the data model needs. The tricky part in my implementation of those two was that I had to keep track of whether the `domain` and `window` were defined by the user or they equal the default value. This part is necessary since the default values of those two parameters can never be set using the constructor, as mentioned above.
```

Comment:
```suggestion
The final task when adding support for a new type in Numba is defining special methods for converting Python objects into native values and vice versa. Those operations are called boxing and unboxing. The idea being that a Python object can be seen as a sophisticated box containing a simple native value which is what the data model needs. The tricky part in my implementation of those two was that I had to keep track of whether the `domain` and `window` were defined by the user or they equal the default value. This part is necessary since the default values of those two parameters can never be set using the constructor, as mentioned above.
```